mlir/lib/Conversion/TosaToRock/TosaToRock.cpp

```diff
@@ -215,6 +215,95 @@ makeRockConv(ConversionPatternRewriter &rw, Operation *op, Value input,
   return cop;
 }
 
+static bool isTosaConv(Operation *op) {
+  return isa<tosa::Conv2DOp, tosa::Conv3DOp, tosa::DepthwiseConv2DOp,
+             tosa::FullyConnectedOp>(op);
+}
+
+static bool isTosaUnary(Operation *op) {
+  return isa<tosa::AvgPool2dOp, tosa::ArgMaxOp, tosa::MaxPool2dOp,
+             tosa::RFFT2dOp, tosa::ClampOp, tosa::SigmoidOp, tosa::TanhOp,
+             tosa::ErfOp, tosa::AbsOp, tosa::BitwiseNotOp, tosa::CeilOp,
+             tosa::ClzOp, tosa::CosOp, tosa::ExpOp, tosa::FloorOp, tosa::LogOp,
+             tosa::LogicalNotOp, tosa::NegateOp, tosa::ReciprocalOp,
+             tosa::RsqrtOp, tosa::SinOp, tosa::PadOp, tosa::ReshapeOp,
+             tosa::ReverseOp, tosa::SliceOp, tosa::TileOp, tosa::TransposeOp,
+             tosa::GatherOp, tosa::ScatterOp, tosa::ResizeOp, tosa::CastOp,
+             tosa::RescaleOp, tosa::IdentityOp>(op);
+}
+
+static bool isTosaBinary(Operation *op) {
+  return isa<
+      tosa::MatMulOp, tosa::FFT2dOp, tosa::AddOp, tosa::ArithmeticRightShiftOp,
+      tosa::BitwiseAndOp, tosa::BitwiseOrOp, tosa::BitwiseXorOp, tosa::IntDivOp,
+      tosa::LogicalAndOp, tosa::LogicalLeftShiftOp, tosa::LogicalRightShiftOp,
+      tosa::LogicalOrOp, tosa::LogicalXorOp, tosa::MaximumOp, tosa::MinimumOp,
+      tosa::MulOp, tosa::PowOp, tosa::SubOp, tosa::TableOp, tosa::EqualOp,
+      tosa::GreaterOp, tosa::GreaterEqualOp>(op);
+}
+
+static bool isTosaReduction(Operation *op) {
+  return isa<tosa::ReduceMaxOp, tosa::ReduceSumOp, tosa::ReduceMinOp,
+             tosa::ReduceProdOp, tosa::ReduceAllOp, tosa::ReduceAnyOp>(op);
+}
+
+static Value traceToRes(Value tensor, func::FuncOp func,
```

Comment:
Why not just iterate on getNumOperands()? 

I think we can remove all of the isXOp() functions.